### PR TITLE
Autocomplete: add StarCoder2 hybrid feature flag

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

- Preparing for the StarCoder2 A/B test.
- Adds `starcoder2-hybrid` model to the Fireworks provider, which uses `starcoder2-7b` and `starcoder2-15b`.
- Cody-Gateway support https://github.com/sourcegraph/sourcegraph/pull/61207

## Test plan

- Tested locally by mocking the return value of the feature flag provider and asserting that the right requests were made by investigating the output console.

### CODE REVIEW

File: vscode/src/completions/providers/fireworks.ts [hunk 1]
```diff
@@ -218,15 +230,27 @@ class FireworksProvider extends Provider {
...
+                      : MODEL_MAP[this.model],
+        } satisfies CodeCompletionsParams
+
+        if (requestParams.model.includes('starcoder2')) {
+            requestParams.stopSequences?.push(
+                '<fim_prefix>',
+                '<fim_suffix>',
+                '<fim_middle>',
+                '<|endoftext|>',
+                '<file_sep>'
```
Comment (philipp-spiess) [inline vscode/src/completions/providers/fireworks.ts:252]: Uhh new `<file_sep>` token 👀  What is this intended for?

**Review by philipp-spiess — APPROVED** [review]

File: vscode/src/completions/providers/fireworks.ts (hunk 1 above)
Comment (valerybugakov) [reply vscode/src/completions/providers/fireworks.ts:252]: Context snippets!

Comment (philipp-spiess) [reply vscode/src/completions/providers/fireworks.ts:252]: Should we use it in the prompt then? 👀

Comment (valerybugakov) [reply vscode/src/completions/providers/fireworks.ts:252]: True. It might improve the suggestions' quality. Let me do it in a separate PR.

**Pull request merged**